### PR TITLE
Made `delegatedIdentityClientId` optional `CustomerManagedKeyEncryption` to align with the common types definition

### LINKED ISSUE #428: [Bug] TypeSpec for CustomerManagedKeyEncryption in Common Type v5 mismatches with corresponding Swagger spec
**Describe the bug**

CustomerManagedKeyEncryption in Common Type v5 defines both Guid properties as **non-nullable**:

```json
"federatedClientId": {
  "type": "string",
  "format": "uuid",
  "description": "application client identity to use for accessing key encryption key Url in a different tenant. Ex: f83c6b1b-4d34-47e4-bb34-9d83df58b540"
},
"delegatedIdentityClientId": {
  "type": "string",
  "format": "uuid",
  "description": "delegated identity to use for accessing key encryption key Url. Ex: /subscriptions/fa5fc227-a624-475e-b696-cdd604c735bc/resourceGroups/<resource group>/providers/Microsoft.ManagedIdentity/userAssignedIdentities/myId. Mutually exclusive with identityType systemAssignedIdentity and userAssignedIdentity - internal use only."
}
```

https://github.com/Azure/azure-rest-api-specs/blob/1d9bdf047d98fb6844ce9309301ca719c1cc003d/specification/common-types/resource-management/v5/customermanagedkeys.json#L42-L53

So does this repo's replica:

https://github.com/Azure/typespec-azure/blob/b10f51eb365962e60674b38f1aca6aa09f68519c/packages/samples/specs/resource-manager/common-types/v5/customermanagedkeys.json#L42-L53

But the corresponding TypeSpec spec defines one of the values as **nullable**:

https://github.com/Azure/typespec-azure/blob/b10f51eb365962e60674b38f1aca6aa09f68519c/packages/typespec-azure-resource-manager/lib/customer-managed-keys.tsp#L39-L43

**Expected behavior**

`federatedClientId` must be redefined as just `federatedClientId: uuid`.

_or_

The Common Types should be updated, what I don't think the case.

### PR DESCRIPTION
**Pull request by abatishchev**
Resolves #428.

### CODE REVIEW

File: .chronus/changes/alexbat/alexbat-delegatedIdentityClientId-1-2024-3-21-20-6.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+changeKind: minor
+packages:
+  - "@azure-tools/typespec-azure-resource-manager"
+---
+
+Made delegatedIdentityClientId a nullable Guid in CustomerManagedKeyEncryptio
```
Comment (timotheeguerin) [inline .chronus/changes/alexbat/alexbat-delegatedIdentityClientId-1-2024-3-21-20-6.md:7]:
```suggestion
Made `delegatedIdentityClientId` optional `CustomerManagedKeyEncryption` to align with the common types definition.
```

**Review by timotheeguerin — APPROVED** [review]
I think this is good to change now as tt should not be breaking the provider hub controller generation as long as the type was generated as a string. With the new change to allow to generate as Guid now the property would be shown differently. 

@markcowl let me know if you agree on that

**Comment by abatishchev** [comment]
@timotheeguerin would I need to submit one more PR to update the pr/providerhub branch?

**Comment by timotheeguerin** [comment]
> @timotheeguerin would I need to submit one more PR to update the pr/providerhub branch?

in what way? this one + your other to allow Guid generation should be good, and next release everthing works

**Comment by abatishchev** [comment]
Sounds good!

The build fails because I crafted the change log file manually. Will fix it on Wednesday (once back home, right now I have only my phone) until you'd have a moment and update my branch with a correct file (idk why exactly it didn't like the one I made).

**Review by markcowl — APPROVED** [review]
This looks good - you will need to regenerate the docs by running a repo-wide build (pnpm build), or you can run the task froo regenerating docs directly for just the ARM library (cd packages/typespec-azure-resource-manager; pnpm regen-docs)

**Comment by abatishchev** [comment]
@markcowl done. and the build is green.

**Pull request merged**